src/_pytest/main.py

```diff
@@ -877,22 +877,24 @@ def resolve_collection_argument(
     If the path doesn't exist, raise UsageError.
     If the path is a directory and selection parts are present, raise UsageError.
     """
-    base, squacket, rest = str(arg).partition("[")
```

Comment:
In the interest of making the change minimal, I would prefer if we keep the same flow as before, i.e. first the `partition("[")` then the `split("::")`. The fix to the issue can be done as an additional step, which is removing the last component of `parts` of it is `""` (this is a known "fixup" to `split()` but which should be explained with a comment).

Reply:
I think this will only solve the case for `test_foo.py::` but not https://github.com/pytest-dev/pytest/issues/10273 E.g. `test_foo.py[foobar]` right?